x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -3812,17 +3945,16 @@ public void testPushdownWithOverwrittenName() {
             assertThat(topN.order().size(), is(2));
 
             var firstOrderExpr = as(topN.order().get(0), Order.class);
-            var mul = as(firstOrderExpr.child(), Mul.class);
-            var add = as(mul.left(), Add.class);
-            var renamed_emp_no = as(add.left(), ReferenceAttribute.class);
-            var renamed_salary = as(add.right(), ReferenceAttribute.class);
-            assertThat(renamed_emp_no.toString(), startsWith("$$emp_no$temp_name"));
-            assertThat(renamed_salary.toString(), startsWith("$$salary$temp_name"));
+            assertThat(firstOrderExpr.direction(), equalTo(org.elasticsearch.xpack.ql.expression.Order.OrderDirection.ASC));
+            assertThat(firstOrderExpr.nullsPosition(), equalTo(org.elasticsearch.xpack.ql.expression.Order.NullsPosition.LAST));
+            var renamedEmpNoSalaryExpression = as(firstOrderExpr.child(), ReferenceAttribute.class);
+            assertThat(renamedEmpNoSalaryExpression.toString(), startsWith("$$order_by$temp_name$0"));
```

Comment:
This doesn't assert the expression that defines `$$order_by$temp_name$0` anymore; I think we should add these asserts, otherwise we only test the correctness of the sort but not of the eval after the opt. rule has run. Same in `testReplaceSortByExpressions`